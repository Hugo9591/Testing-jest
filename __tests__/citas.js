import Citas from '../js/classes/Citas';

describe('Probar la clase de citas', ()=>{

    const citas = new Citas();
    const id = Date.now();

    test('Agregar nueva cita', ()=>{
        const citaObj = {
            id,
            mascota: 'Hook',
            propietario: 'Hugo',
            telefono: '1234567',
            fecha: '10-12-24',
            hora:'10:30',
            sintomas: 'solo duerme poco'
        };

        citas.agregarCita(citaObj);

        //Prueba
        expect(citas).toMatchSnapshot();
        //Nos crea una carpeta y un archivo con la informaacion dada para despues ser usada para comparar con alguna actualizacion
    });



    test('Actualizar cita', ()=>{
        const citaActualizada = {
            id,
            mascota: 'Nuevo Nombre',
            propietario: 'Hugo',
            telefono: '1234567',
            fecha: '10-12-24',
            hora:'10:30',
            sintomas: 'solo duerme'
        };

        citas.editarCita(citaActualizada);

        expect(citas).toMatchSnapshot();
    });


    test('Eliminar cita',()=>{
        citas.eliminarCita(id);

        expect(citas).toMatchSnapshot();
    });
})