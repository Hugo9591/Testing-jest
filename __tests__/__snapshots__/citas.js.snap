// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de citas Actualizar cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-24",
      "hora": "10:30",
      "id": 1740460482700,
      "mascota": "Nuevo Nombre",
      "propietario": "Hugo",
      "sintomas": "solo duerme",
      "telefono": "1234567",
    },
  ],
}
`;

exports[`Probar la clase de citas Agregar nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-24",
      "hora": "10:30",
      "id": 1740460482700,
      "mascota": "Hook",
      "propietario": "Hugo",
      "sintomas": "solo duerme poco",
      "telefono": "1234567",
    },
  ],
}
`;

exports[`Probar la clase de citas Eliminar cita 1`] = `
Citas {
  "citas": [],
}
`;
